Sources/SWBCSupport/CLibclang.cpp

```diff
@@ -2159,6 +2170,22 @@ extern "C" {
         return depGraph != nullptr;
     }
 
+    bool libclang_scanner_generate_reproducer(libclang_scanner_t scanner,
+                                              int argc, char *const *argv,
+                                              const char *workingDirectory,
+                                              const char **message) {
+        auto lib = scanner->scanner->lib;
+        LibclangFunctions::CXString messageString;
+        auto result = lib->fns.clang_experimental_DependencyScanner_generateReproducer(
+            argc, const_cast<const char**>(argv), workingDirectory,
+            /*ReproducerLocation=*/NULL, /*UseUniqueReproducerName=*/true, &messageString);
+        if (message) {
+            *message = strdup(lib->fns.clang_getCString(messageString));
```

Comment:
Is libclang guaranteed to return a non null string in messageString here?

Reply:
For success case I think we can make it mandatory for messageString to be present (you need to tell where to find the generated reproducer). Error cases might not populate messageString. Even though they should provide more details, it is fairly easy to return InvalidArgument error without any extra message.

But we can guarantee messageString isn't corrupted and in the worst case it would be an empty string.

I'll think about these cases and would document API contract on libclang side.